Cython/Utility/Builtins.c

```diff
@@ -312,17 +312,21 @@ static CYTHON_INLINE PyObject* __Pyx_divmod_int(int a, int b) {
     // This results in different answers between Python and C/C++
     // when the dividend is negative and the divisor is positive and vice versa.
     int q, r;
-    if ((a < 0 && b > 0) || (a > 0 && b < 0)) {
+    if (unlikely(b == 0)) {
+        PyErr_SetString(PyExc_ZeroDivisionError, "integer division or modulo by zero");
+        return NULL;
+    }
+    else if (a == 0) {
+        q = 0;
+        r = 0;
+    }
+    else if ((a > 0) != (b > 0)) {
```

Comment:
`b == 0` was probably tested last because it's an exceptional case, to make sure that we jump in the fast path immediately. Can't say how important `a == 0` is. Probably not something to invest time on in the normal case.

Reply:
OTOH, your ordering seems perfectly reasonable and fast to me. I'm fine with it.